### DISCUSSION TITLE
Turn off autocomplete for csrf_field generates HTML validation error

### DISCUSSION THREAD

**Discussion by Graphicfort** [laravel/framework#48842] [General]
**An “input” element with a “type” attribute whose value is “hidden” must not have an “autocomplete” attribute whose value is “on” or “off”**

To avoid HTML5 validation error, you can change:
<pre>
return new HtmlString('&lt;input type="<b>hidden</b>" name="_token" value="'.csrf_token().'" autocomplete="off"&gt;');
</pre>
to
<pre>
return new HtmlString('&lt;input type="<b>text</b>" name="_token" value="'.csrf_token().'" autocomplete="off" <b>style="display:none !important;"</b>&gt;');
</pre>

**Comment by valorin** [comment]
Inline styles could cause issues with Content Security Policies, so it would be better to just remove the attribute entirely. 
It was added as a fix for a potential browser issue that turned out to be a different issue.

**Reply by Graphicfort** [reply]
I'm aware of safari issue. The problem I have a table with 50+ form so I'm getting 50+ validation errors, I ended up creating my own csrf field  with a class to hide it, but this is not the best solution.

**Reply by valorin** [reply]
The Safari issue seemed to be that the page was being cached, so an older CSRF token was being submitted: https://github.com/laravel/framework/pull/48371#issuecomment-1719852125

Inline styles are not a good solution. If it's causing issues, the attribute should be removed.
